src/main/java/Bob.java

```diff
@@ -0,0 +1,338 @@
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.Path;
+import org.apache.commons.lang3.StringUtils;
+
+public class Bob {
+    private final Scanner scanner = new Scanner(System.in);
+
+    private ArrayList<Task> taskList = new ArrayList<>();
+    private static final Integer spacing = 5;
+
+    private static final String wrapper = padLeft("~".repeat(30));
+
+    private static final DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd[ ha]");
+
+    private static String padLeft (String s) {
+        return " ".repeat(spacing) + s;
+    }
+
+    private final Path dataDirectoryPath = Paths.get("data/");
+    private final Path dataFilePath = Paths.get("data/taskList.txt");
+
+    // Accepts strings that can be separated by \n
+    private void formattedPrint(String s) {
+        String[] lines = s.split("\n");
+        System.out.println(wrapper);
+        for (String line : lines) {
+            System.out.println(padLeft(line));
+        }
+        System.out.println(wrapper);
+    }
+
+    private String getTaskDescription(Task t) {
+        return String.format("[%s][%s] %s", t.getTaskType(), t.getStatusIcon(), t);
+    }
+    private void printList() {
+        System.out.println(wrapper);
+        System.out.println(padLeft("Current task list: "));
+        // Iterate through list items sequentially
+        for (int i = 0, n = this.taskList.size(); i < n; i++) {
+            Task t = this.taskList.get(i);
+            String s = String.format("%d. %s", i + 1, getTaskDescription(t));
+            System.out.println(padLeft(s));
+        }
+        System.out.println(wrapper);
+    }
+
+    //Check that the string is a number
+    private static boolean isInt(String s) {
+        if (s == null) {
+            return false;
+        }
+        // Check that every char is a digit
+        for (int i = 0, len = s.length(); i < len; i++) {
+            char c = s.charAt(i);
+            if (c < '0' || c > '9') {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    // Check if string can be parsed to LocalDate
+    private static boolean isDate(String s) {
+        try {
+            LocalDate.parse(s, format);
+            return true;
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+    }
+    private void handleMarkCommand(String s) throws BobException{
```

Comment:
Should there be no space after BobException?

Reply:
Thanks for catching this!